test/tests/listbox-collapsible.js

```diff
@@ -0,0 +1,364 @@
+'use strict';
+
+const { ariaTest } = require('..');
+const { By, Key } = require('selenium-webdriver');
+const assertAttributeValues = require('../util/assertAttributeValues');
+const assertAttributeDNE = require('../util/assertAttributeDNE');
+const assertAriaRoles = require('../util/assertAriaRoles');
+const assertAriaLabelledby = require('../util/assertAriaLabelledby');
+const assertAriaSelectedAndActivedescendant = require('../util/assertAriaSelectedAndActivedescendant');
+
+const exampleFile = 'listbox/listbox-collapsible.html';
+
+const ex = {
+  buttonSelector: '#ex button',
+  listboxSelector: '#ex [role="listbox"]',
+  optionSelector: '#ex [role="option"]',
+  numOptions: 26
+};
+
+const checkFocus = async function (t, selector) {
+  return await t.context.session.wait(async function () {
+    return t.context.session.executeScript(function () {
+      const [ selector ] = arguments;
+      const items = document.querySelector(selector);
+      return items === document.activeElement;
+    }, selector);
+  }, t.context.waitTime, 'Timeout waiting for dom focus on element:' + selector);
+};
+
+ariaTest('"aria-labelledby" on button element', exampleFile, 'button-aria-labelledby', async (t) => {
+  t.plan(1);
+  await assertAriaLabelledby(t, 'ex', ex.buttonSelector);
+});
+
+
+ariaTest('"aria-haspopup" on button element', exampleFile, 'button-aria-haspopup', async (t) => {
+  t.plan(1);
+  await assertAttributeValues(t, ex.buttonSelector, 'aria-haspopup', 'listbox');
+});
+
+ariaTest('"aria-expanded" on button element', exampleFile, 'button-aria-expanded', async (t) => {
+  t.plan(4);
+
+  const button = await t.context.session.findElement(By.css(ex.buttonSelector));
+
+  // Check that aria-expanded is not set and the listbox is not visible before interacting
+
+  await assertAttributeDNE(t, ex.buttonSelector, 'aria-expanded');
+
+  t.false(
+    await t.context.session.findElement(By.css(ex.listboxSelector)).isDisplayed(),
+    'listbox element should not be displayed when \'aria-expanded\' does not exist'
+  );
+
+  // click button
```

Comment:
Is this meant to be turned into an action later?

Reply:
it just describes the next line of code :)